src/api/app/models/concerns/staging_project.rb

```diff
@@ -142,35 +142,26 @@ def cache_problems
   end
 
   def state
-    return :empty unless staged_requests.exists?
-    return :unacceptable if unacceptable_state?
-    return :review if review_state?
-    general_state
-  end
-
-  def unacceptable_state?
-    untracked_requests.present? || staged_requests.obsolete.present?
-  end
-
-  def review_state?
-    general_state == :acceptable && missing_reviews.present?
+    return :empty if staged_requests.blank?
+    return :unacceptable if untracked_requests.present? || staged_requests.obsolete.present?
+    return build_and_check_state if build_and_check_state
```

Comment:
This method is being called twice

Reply:
You are right! Something similar happened inside `build_and_check_state` so we have changed all a bit.